tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/annotation/MetaAttributeGenerationVisitor.java

```diff
@@ -138,6 +138,22 @@ private AnnotationMetaAttribute createMetaCollectionAttribute(DeclaredType decla
 				accessTypeInfo.setDefaultAccessType( entity.getEntityAccessTypeInfo().getAccessType() );
 			}
 		}
+		if ( TypeUtils.containsAnnotation(
+				element,
+				Constants.CONVERT,
+				Constants.HIBERNATE_TYPE
+		) && !TypeUtils.containsAnnotation(
+				element,
+				Constants.ONE_TO_MANY,
+				Constants.MANY_TO_MANY,
+				Constants.ELEMENT_COLLECTION
+		) ) {
+			return new AnnotationMetaSingleAttribute(
+					entity,
+					element,
+					TypeUtils.toTypeString( declaredType )
+			);
+		}
```

Comment:
I don't think this is according to the JPA specification, but I didn't look into it to find a reference. My gut tells me, this should be fixed by annotating the target attribute with `@Basic` so IMO it should look like the following:

```suggestion
		if ( TypeUtils.containsAnnotation(
				element,
				Constants.CONVERT,
				Constants.HIBERNATE_TYPE,
				Constants.BASIC
		) ) {
			return new AnnotationMetaSingleAttribute(
					entity,
					element,
					TypeUtils.toTypeString( declaredType )
			);
		}
```